npm/pkg/dataplane/policies/chain-management_linux.go

```diff
@@ -0,0 +1,397 @@
+	creatorToFlush, chainsToDelete := pMgr.getCreatorAndChainsForReset()
+	restoreError := restore(creatorToFlush)
+	if restoreError != nil {
+		return npmerrors.SimpleErrorWrapper("failed to flush chains", restoreError)
+	}
+
+	// TODO aggregate an error for each chain that failed to delete
+	var anyDeleteErr error
+	for _, chainName := range chainsToDelete {
+		errCode, err := pMgr.runIPTablesCommand(util.IptablesDestroyFlag, chainName)
+		if err != nil {
+			klog.Infof("couldn't delete chain %s with error [%v] and exit code [%d]", chainName, err, errCode)
+			anyDeleteErr = err
+		}
+	}
+
+	if anyDeleteErr != nil {
+		return npmerrors.SimpleErrorWrapper("couldn't delete all chains", anyDeleteErr)
+	}
+	return nil
+}
+
+// ReconcileChains periodically creates the jump rule from FORWARD chain to AZURE-NPM chain (if it d.n.e)
+// and makes sure it's after the jumps to KUBE-FORWARD & KUBE-SERVICES chains (if they exist).
+func (pMgr *PolicyManager) ReconcileChains(stopChannel <-chan struct{}) {
+	go pMgr.reconcileChains(stopChannel)
+}
+
+func (pMgr *PolicyManager) reconcileChains(stopChannel <-chan struct{}) {
+	ticker := time.NewTicker(time.Minute * time.Duration(reconcileChainTimeInMinutes))
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-stopChannel:
+			return
+		case <-ticker.C:
+			if err := pMgr.positionAzureChainJumpRule(); err != nil {
+				metrics.SendErrorLogAndMetric(util.NpmID, "Error: failed to reconcile jump rule to Azure-NPM due to %s", err.Error())
+			}
+		}
+	}
+}
+
+// this function has a direct comparison in NPM v1 iptables manager (iptm.go)
+func (pMgr *PolicyManager) runIPTablesCommand(operationFlag string, args ...string) (int, error) {
+	allArgs := []string{util.IptablesWaitFlag, defaultlockWaitTimeInSeconds, operationFlag}
+	allArgs = append(allArgs, args...)
+
+	if operationFlag != util.IptablesCheckFlag {
+		klog.Infof("Executing iptables command with args %v", allArgs)
+	}
+
+	command := pMgr.ioShim.Exec.Command(util.Iptables, allArgs...)
+	output, err := command.CombinedOutput()
+
+	var exitError utilexec.ExitError
+	if ok := errors.As(err, &exitError); ok {
+		errCode := exitError.ExitStatus()
+		allArgsString := strings.Join(allArgs, " ")
+		msgStr := strings.TrimSuffix(string(output), "\n")
+		if errCode > 0 && operationFlag != util.IptablesCheckFlag {
+			metrics.SendErrorLogAndMetric(util.IptmID, "Error: There was an error running command: [%s %s] Stderr: [%v, %s]", util.Iptables, allArgsString, exitError, msgStr)
+		}
+		return errCode, npmerrors.SimpleErrorWrapper(fmt.Sprintf("failed to run iptables command [%s %s] Stderr: [%s]", util.Iptables, allArgsString, msgStr), exitError)
+	}
+	return 0, nil
+}
+
+func (pMgr *PolicyManager) getCreatorForInitChains() *ioutil.FileCreator {
+	creator := pMgr.getNewCreatorWithChains(iptablesAzureChains)
+
+	// add AZURE-NPM chain rules
+	creator.AddLine("", nil, util.IptablesAppendFlag, util.IptablesAzureChain, util.IptablesJumpFlag, util.IptablesAzureIngressChain)
+
```

Comment:
nitPick: remove unneeded newlines in this function if there is no specific reason.